core/src/main/scala/org/apache/spark/HttpServer.scala

```diff
@@ -155,6 +156,7 @@ private[spark] class HttpServer(
       throw new ServerStateException("Server is already stopped")
     } else {
       server.stop()
+      Option(server.getThreadPool).collect { case x: LifeCycle => x }.foreach(_.stop())
```

Comment:
btw in this case the option, colllect, foreach isn't that bad. but i often find people learn from these examples and go overboard with chaining. i constantly need to remind contributors to simplify their use of scala features. most recently

https://github.com/apache/spark/pull/12256/files#r58988046